### ISSUE TITLE
Utilizing Soundcloud SDK in Ember App

### ISSUE THREAD

**Issue by kpie89** [ga-wdi-boston/capstone-project#108]
I'm unclear how to utilize this soundcloud module. I have installed it locally on my project but it doesn't recognize when I try to import it into my soundcloud ember service. 

https://github.com/soundcloud/soundcloud-javascript

There's the official repo for the SDK. 

https://developers.soundcloud.com/docs/api/sdks#javascript

There is the Docs. 

It specifically says "You can also install the SDK via NPM now: npm install soundcloud." 

I don't really understand how modules work with Ember. Or if they do?

**Comment by jrhorn424** [comment]
Did you `npm install soundcloud --save-dev`? Afterward, you'll likely need to do something to `ember-cli-build.js` to add the soundcloud module to your application bundle. See, for example, how we've included the default bootstrap javascript module for Ember in the template (`app.import`).

You should then be able to use it as intended in the docs.

**Comment by kpie89** [comment]
So with bootstrap you import "'bower_components/bootstrap-sass/assets/javascripts/bootstrap.js’”…where exactly does this live? So I can have a better idea where my new souncloud module lives

**Comment by jrhorn424** [comment]
After you `bower install`, `bower_components` exists in the project directory. After you `npm install`, `node_modules` exists in the project directory. I'd check inside `node_modules`. 😉

**Comment by kpie89** [comment]
Thank you. It was indeed inside node_modules

**Comment by jrhorn424** [comment]
You'll need to use an Ember initializer to perform the initial authentication with soundcloud.
1. `ember generate intiailzer soundlcoud`
2. Import and use `SC.initialize()` in the main initializer function.

**Comment by jrhorn424** [comment]
@kpie89 ran into an issue where using `node_modules` in an `app.import` was causing the build to fail.

I initially saw a [suggestion to use `ember-browserify`](http://stackoverflow.com/a/29544299/402618). I ignored this earlier since I viewed [`browserify`](http://browserify.org) as an alternative to [Broccoli](http://broccolijs.com).

It turns out [`ember-browserify`](https://github.com/ef4/ember-browserify) is unrelated to the build tool of the same name. `ember-browserify` just allows you to load `npm` modules in your application.

``` sh
npm install foo --save-dev
```

``` js
// some part of Ember
import Foo from 'npm:foo';
```

The recommendation is to still use `bower` and `app.import` when available, but if a package is only available on `npm`, it can be loaded with `ember-browserify`.
